pkg/auth/oauth/handlers/default_auth_handler.go

```diff
@@ -113,6 +115,21 @@ func (authHandler *unionAuthenticationHandler) AuthenticationNeeded(apiClient au
 				w.WriteHeader(http.StatusFound)
 			default:
 				w.WriteHeader(http.StatusUnauthorized)
+				ctx, ok := authHandler.requestContextMapper.Get(req)
+				if !ok {
+					return false, fmt.Errorf("no context found for request to audit")
+				}
+				ev := request.AuditEventFrom(ctx)
+				if ev != nil {
+					// this code mimics the bits from k8s.io/apiserver/pkg/endpoints/filters/authn_audit.go
+					// but since we don't accept failedHander here we need to manually alter the audit
+					// event with information about failed authentication
+					ev.ResponseStatus.Message = getAuthMethods(req)
+					ctx = request.WithAuditEvent(ctx, ev)
+					if err := authHandler.requestContextMapper.Update(req, ctx); err != nil {
+						return false, fmt.Errorf("failed to attach audit event to context: %v", err)
```

Comment:
Why does this error?

Reply:
Similar to above case where we get context, after we modify the context with new event we need to post that back. If that fails, we're loosing information added here.